packages/labs/ssr/src/lib/lit-element-renderer.ts

```diff
@@ -29,6 +33,31 @@ export class LitElementRenderer extends ElementRenderer {
   constructor(tagName: string) {
     super(tagName);
     this.element = new (customElements.get(this.tagName)!)() as LitElement;
+
+    // Reflect internals AOM attributes back to the DOM prior to hydration to
+    // ensure search bots can accurately parse element semantics prior to
+    // hydration. This is called whenever an instance of ElementInternals is
+    // created on an element to wire up the getters/setters for the ARIAMixin
+    // properties.
+    const internals = (
+      this.element as object as {__internals: ElementInternals}
```

Comment:
So this only work when we're using our shim. We need to think about what happens if someone loads jsdom. We may need adapters, like a `getInternals()` callback.

Reply:
We should have an issue somewhere for using another DOM shim, and this can go on the list of things we'll need to consider there. What I mean by `getInternals()` is that since there's no standard DOM API for getting at that from outside a component, we'll need a callback per DOM emulation library to get at it. Future work, just noting it.